chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/derivation/transformer/rules/TransformSealedHierarchyToSealedHierarchyRuleModule.scala

```diff
@@ -107,7 +107,9 @@ private[compiletime] trait TransformSealedHierarchyToSealedHierarchyRuleModule {
         ctx: TransformationContext[From, To]
     ): DerivationResult[Existential[ExprPromise[*, TransformationExpr[To]]]] = {
       import fromSubtype.Underlying as FromSubtype, fromSubtype.value.name as fromName
-      toElements.filter(toSubtype => enumNamesMatch(fromName, toSubtype.value.name)).toList match {
+      toElements.filter(toSubtype =>
+        ctx.config.flags.subtypeNameComparison.namesMatch(fromName, toSubtype.value.name)
```

Comment:
I guess `TransformerContext` might introduce some utilities so that these would become `areSubtypeNamesEqual(fromName, toName)` where the utility would delegate to flags.subtypeNameComparison.namesMatch.

Alternatively, to not require importing this utility everywhere could be:

```scala
def areSubtypeNamesMatching(fromName: String, toName: String)(
  implicit ctx: TransformationContext[?, ?]
): Boolean =
  ctx.config.flags.subtypeNameComparison.namesMatch(fromName, toName)
```

defined in `Configurations`.

Reply:
Resolved on #478